editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,322 @@
+					getWidgetFactory()).create();
+		}
+
+		if (txtFieldCreator != null) {
+			ISWTObservableValue uiObservable = handleObservable(txtFieldCreator, p);
+			if (uiObservable != null) {
+				bindField(validator, uiObservable, p);
+			}
+		}
+	}	
+	
+	private ISWTObservableValue handleObservable(AbstractTextFieldParameterPropertyUICreator txtFieldCreator, Parameter p) {
+		// initialize the map entry
+		modelMap.put(p.getName(), txtFieldCreator.getControl().getText());
+		// create observables for the control
+		return txtFieldCreator.getUiObservable();
+	}
+	
+	private void bindField(IValidator validator, ISWTObservableValue uiObservable, Parameter p) {
+		// create UpdateValueStrategy and assign to the binding
+		UpdateValueStrategy strategy = new UpdateValueStrategy();
+		strategy.setBeforeSetValidator(validator);
+
+		// create observables for the Map entries
+		IObservableValue<Object> modelObservable = Observables.observeMapEntry(modelMap, p.getName());
+		// bind the observables
+		Binding bindValue = dbc.bindValue(uiObservable, modelObservable, strategy, null);
+		ControlDecorationSupport.create(bindValue, SWT.TOP | SWT.LEFT);
+	}
+	
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.ui.views.properties.tabbed.AbstractPropertySection#
+	 * createControls (org.eclipse.swt.widgets.Composite,
+	 * org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetPage)
+	 */
+	@Override
+	public void createControls(Composite parent, TabbedPropertySheetPage aTabbedPropertySheetPage) {
+		this.toolkit = new FormToolkit(parent.getDisplay());
+		super.createControls(parent, aTabbedPropertySheetPage);
+
+		// now setup the file binding properties page
+		parent.setLayout(new GridLayout());
+		parent.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+		form = toolkit.createForm(parent);
+		form.setLayoutData(new GridData(GridData.FILL_BOTH));
+		form.getBody().setLayout(new GridLayout(1, false));
+
+		Composite sbody = form.getBody();
+
+		tabFolder = new CTabFolder(sbody, SWT.TOP | SWT.FLAT);
+		toolkit.adapt(tabFolder, true, true);
+		tabFolder.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+		Color selectedColor = toolkit.getColors().getColor(IFormColors.SEPARATOR);
+		tabFolder.setSelectionBackground(new Color[] { selectedColor, toolkit.getColors().getBackground() },
+				new int[] { 20 }, true);
+		tabFolder.setCursor(FormsResources.getHandCursor());
+		toolkit.paintBordersFor(tabFolder);
+
+		form.setText("Advanced Properties");
+		toolkit.decorateFormHeading(form);
+
+		form.layout();
+		tabFolder.setSelection(0);
+	}
+
+	private void createClassBrowseButton(Composite composite, Text field) {
+		Button browseBeanButton = new Button(composite, SWT.PUSH);
+		browseBeanButton.setText("...");
+		browseBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				final IProject project = CamelUtils.project();
```

Comment:
better to avoid to use the CamelUtils.project() as it is linking this code the workbench state.
On AbstractCamelElement we can getCamelFile().getResource().getProject()